File: swing/src/main/java/wannabe/swing/renderer/FilledSquareWithCabinetWires.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.Projected;

public class FilledSquareWithCabinetWires extends SwingRenderer {
  Sides sides = new Sides();
  @Override public void draw(Graphics g, Projected r) {
    sides.wireSides(g, r);
    g.fillRect(r.left, r.top, r.size, r.size);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/FilledSquareWithCabinetWires.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class FilledSquareWithCabinetWires : SwingRenderer() {
  private var sides = Sides()

  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    sides.wireSides(g, p)
    g.fillRect(p.left, p.top, p.size, p.size)
  }
}
